test/Gen.hs

```diff
@@ -61,26 +76,28 @@ import Hedgehog.Range
 
 -- | A generator for computations, given a higher-order generator for effectful operations, & a generator for results.
 m
-  :: forall m a
+  :: forall m
   .  Monad m
-  => (forall a . (forall a . Gen a -> Gen (m a)) -> Gen a -> Gen (m a)) -- ^ A higher-order generator producing operations using any effects in @m@.
-  -> Gen a                                                              -- ^ A generator for results.
-  -> Gen (m a)                                                          -- ^ A generator producing computations.
-m with = go where
-  go :: forall a . Gen a -> Gen (m a)
-  go a = Gen $ recursive Hedgehog.Gen.choice
+  => (GenM m -> GenM m) -- ^ A higher-order computation generator using any effects in @m@.
+  -> GenM m             -- ^ A computation generator.
+m with = m where
+  m :: GenM m
+  m a = Gen $ scale (`div` 2) $ recursive Hedgehog.Gen.choice
```

Comment:
We now scale down recursive constructions in general to avoid constructing infinite terms for nondeterministic carriers (which have a potentially O(2ⁿ) constructor in `<|>`) in the `MonadFix` tests. I think this is a pretty good step in general here too, because it’s generally the case that large computations aren’t as interesting as ones with high variance; so we’re better off running the tests ten thousand times than running the tests over dramatically larger computations—especially if “dramatically larger” means “infinite larger” 😂